sql-plugin/src/main/scala/org/apache/spark/sql/rapids/RapidsShuffleInternalManagerBase.scala

```diff
@@ -636,6 +636,14 @@ abstract class RapidsShuffleThreadedReaderBase[K, C](
       null
     }
 
+    // Register a completion handler to close any queued cbs.
+    Option(TaskContext.get()).foreach {_.addTaskCompletionListener[Unit]( _ => {
```

Comment:
@abellina can I just use the `context` from `RapidsShuffleThreadedReaderBase` here?  Same applies for line 550.

Reply:
yes this is the same `TaskContext` as what is in `TaskContext.get`

Reply:
Thanks. I will update this.